design/003-interactive-session/003-interactive-session.md

```diff
@@ -0,0 +1,94 @@
+- Start Date: 2021-12-02
+- Status: Proposed
+
+# Run local interactive sessions
+
+## Summary
+
+Propose a new sub-group command `renku session` that is responsible for
+managing local interactive sessions for a project.
+
+## Motivation
+
+Renkulab.io provides the option to run interactive sessions of a project. Among other things the session
+contains a jupyter notebook.
+
+The motivation behind this RFC is to provide a simple renku CLI command that basically mimics this, i.e. the
+option to start an interactive session of a project locally.
+
+## Design Detail
+
+The proposed sub-commands of `renku session`, namely to run interactive sessions locally, heavily relies on
+[docker](https://www.docker.com/). Docker is one of the most popular container engines, that eases the development
+and deployment of applications, by off-loading the installation of the runtime environment dependencies of the
+application.
+
+In order to start a local interactive session, just as in case of renkulab.io, it is required to provide
+a `Dockerfile`. This is by default provided by `renku init` when creating a renku project, but if the user
+is using a custom project template it might not be present.
+
+### Proposed Commands
+
+#### renku session start
+
+A command to start an interactive session.
+
+By default, the `Dockerfile` present in the given project is going to be used for building the docker image
+that's going to be used for running the interactive session. In case the docker image is not available the
+user is going to be prompted whether the user wants to build the docker image, as in some cases it can be
+time consuming.
```

Comment:
What do we do in the case of a private project? We _should_ be able to get access to the hosted gitlab registry with the token we get from `renku login` but I'm not 100% sure. This has to be tested/investigated since I believe it's a JWT token which is replaced by an oauth token by the gateway - but I don't know if this works for the registry endpoints or not. It might require changes to the gateway.